Custom Climate component is "Unavailable" in Home Assistant after upgrade to 1.19.0

<!-- Thanks for reporting a bug for this project. READ THIS FIRST:

- Provide as many details as possible. Simply saying "X gives bug" or "X gives error" is not enough!
- Paste logs, configuration sample and code into the backticks (```).
- Read through the template carefully and fill out all missing details.
- Please also search for similar issues in this issue tracker first and read through the ESPHome FAQ.

DO NOT DELETE ANY TEXT from this template! Otherwise the issue may be closed without a comment.
-->

**Operating environment/Installation (Hass.io/Docker/pip/etc.):**
<!--
Please provide details about your environment below this line. -->
HassOS, Raspberry Pi 3B+

**ESP (ESP32/ESP8266, Board/Sonoff):**
<!--
Please provide details about which ESP you're using below.
-->
NodeMCU

**ESPHome version (latest production, beta, dev branch)**
<!--
ESPHome version like v1.14 or 1.15-dev
-->
1.19.0

**Affected component:**
<!--
Please add the link to the documentation at https://esphome.io/index.html of the component in question.
-->
https://esphome.io/components/climate/custom.html

**Description of problem:**
Ever since the upgrade to 1.19.0 the entity is "Unavailable" in Home Assistant and Exception is thrown and displayed in home-assistant.log

**Problem-relevant YAML-configuration entries:**
```yaml
esphome:
  name: mcu_dnevna_soba_transciever
  platform: ESP8266
  board: nodemcuv2
  includes: 
    - custom_components/gree/gree_ir.h
  libraries:
    - IRremoteESP8266

wifi:
  ssid: "SSID"
  password: "xxx"

  # Enable fallback hotspot (captive portal) in case wifi connection fails
  ap:
    ssid: "MCU Dnevna Soba Transciever"
    password: "xxx"

captive_portal:

# Enable logging
logger:
  level: DEBUG

# Enable Home Assistant API
api:
  password: "xxx"

ota:
  password: "xxx"

mqtt:
  broker: 192.168.0.201
  username: user
  password: pass
  discovery: false
  discovery_retain: false

remote_transmitter:
  pin: 
    number: D0
    inverted: false
  carrier_duty_percent: 50%
  id: "ir_dsoba_transmitter"

climate:
- platform: custom
  lambda: |-
    auto dsoba_ac = new GreeAC();
    App.register_component(dsoba_ac);
    return {dsoba_ac};
  climates:
    - name: "Dnevna Soba AC"

switch:
  - platform: template
    id: mi_tv_pwr
    name: "Mi TV"
    lambda: |-
      if (id(mi_tv_status).state) {
        return true;
      } else {
        return false;
      }
    turn_on_action:
     - remote_transmitter.transmit_raw:
         code: [1004, -581, 581, -582, 581, -1504, 557, -1504, 557, -582, 581, -1505, 556, -582, 580, -1506, 555, -583, 581, -1504, 556, -1505, 556, -10752, 1006, -584, 578, -581, 583, -1504, 557, -1504, 557, -606, 556, -1505, 578, -561, 580, -1503, 557, -607, 579, -1482, 557, -1503, 557, -10753, 1006, -583, 602, -557, 584, -1503, 582, -1479, 581, -582, 580, -1481, 581, -557, 608, -1477, 581, -556, 608, -1479, 582, -1479, 583, -10726, 1031, -557, 607, -555, 607, -1479, 582, -1479, 583, -553, 610, -1478, 583, -555, 608, -1477, 607, -530, 610, -1476, 584, -1477, 585, -10725, 1058, -528, 633, -529, 612, -1474, 586, -1476, 585, -552, 610, -1476, 586, -552, 634, -1452, 608, -529, 634, -1453, 607, -1453, 586, -10723, 1035, -552, 634, -529, 634, -1427, 634, -1428, 632, -529, 633, -1428, 633, -529, 634, -1427, 633, -530, 633, -1427, 634, -1427, 634, -10702, 1056, -530, 631, -531, 631, -1430, 632, -1429, 632, -531, 630, -1430, 631, -532, 631, -1430, 630, -533, 629, -1431, 629, -1432, 629, -10707, 1050, -535, 603, -560, 603, -1458, 603, -1458, 603, -560, 602, -1458, 603, -560, 603, -1458, 602, -561, 601, -1460, 601, -1460, 600, -10737, 1021, -564, 599, -564, 598, -1483, 578, -1483, 578, -584, 578, -1483, 577, -587, 576, -1484, 576, -588, 575, -1485, 575, -1487, 574, -10762, 996, -590, 571, -592, 571, -1490, 570, -1491, 568, -595, 567, -1495, 566, -597, 565, -1497, 564, -598, 565, -1519, 542, -1519, 542, -10795, 964, -621, 541, -621, 542, -1519, 542, -1519, 542, -621, 541, -1520, 542, -621, 541, -1520, 542, -620, 542, -1519, 543, -1519, 540, -10795, 965, -621, 541, -622, 541, -1520, 542, -1519, 541, -621, 542, -1520, 540, -623, 540, -1521, 539, -624, 516, -1545, 516, -1569, 492, -10844, 914, -671, 491, -672, 490, -1572, 490, -1570, 491, -672, 490, -1572, 489, -673, 490, -1571, 490, -673, 490, -1587, 473, -1597, 464, -10873, 885, -701, 461, -726, 437, -1624, 437, -1625, 436, -727, 435, -1626, 435, -727, 411, -1676, 385, -778, 384, -1677, 384, -1704, 357, -11322, 222]
         carrier_frequency: 38kHz
    turn_off_action:
     - remote_transmitter.transmit_raw:
         code: [1004, -581, 581, -582, 581, -1504, 557, -1504, 557, -582, 581, -1505, 556, -582, 580, -1506, 555, -583, 581, -1504, 556, -1505, 556, -10752, 1006, -584, 578, -581, 583, -1504, 557, -1504, 557, -606, 556, -1505, 578, -561, 580, -1503, 557, -607, 579, -1482, 557, -1503, 557, -10753, 1006, -583, 602, -557, 584, -1503, 582, -1479, 581, -582, 580, -1481, 581, -557, 608, -1477, 581, -556, 608, -1479, 582, -1479, 583, -10726, 1031, -557, 607, -555, 607, -1479, 582, -1479, 583, -553, 610, -1478, 583, -555, 608, -1477, 607, -530, 610, -1476, 584, -1477, 585, -10725, 1058, -528, 633, -529, 612, -1474, 586, -1476, 585, -552, 610, -1476, 586, -552, 634, -1452, 608, -529, 634, -1453, 607, -1453, 586, -10723, 1035, -552, 634, -529, 634, -1427, 634, -1428, 632, -529, 633, -1428, 633, -529, 634, -1427, 633, -530, 633, -1427, 634, -1427, 634, -10702, 1056, -530, 631, -531, 631, -1430, 632, -1429, 632, -531, 630, -1430, 631, -532, 631, -1430, 630, -533, 629, -1431, 629, -1432, 629, -10707, 1050, -535, 603, -560, 603, -1458, 603, -1458, 603, -560, 602, -1458, 603, -560, 603, -1458, 602, -561, 601, -1460, 601, -1460, 600, -10737, 1021, -564, 599, -564, 598, -1483, 578, -1483, 578, -584, 578, -1483, 577, -587, 576, -1484, 576, -588, 575, -1485, 575, -1487, 574, -10762, 996, -590, 571, -592, 571, -1490, 570, -1491, 568, -595, 567, -1495, 566, -597, 565, -1497, 564, -598, 565, -1519, 542, -1519, 542, -10795, 964, -621, 541, -621, 542, -1519, 542, -1519, 542, -621, 541, -1520, 542, -621, 541, -1520, 542, -620, 542, -1519, 543, -1519, 540, -10795, 965, -621, 541, -622, 541, -1520, 542, -1519, 541, -621, 542, -1520, 540, -623, 540, -1521, 539, -624, 516, -1545, 516, -1569, 492, -10844, 914, -671, 491, -672, 490, -1572, 490, -1570, 491, -672, 490, -1572, 489, -673, 490, -1571, 490, -673, 490, -1587, 473, -1597, 464, -10873, 885, -701, 461, -726, 437, -1624, 437, -1625, 436, -727, 435, -1626, 435, -727, 411, -1676, 385, -778, 384, -1677, 384, -1704, 357, -11322, 222]
         carrier_frequency: 38kHz
  - platform: template
    id: mi_tv_up
    name: "Mi TV Up"
    turn_on_action:
     - remote_transmitter.transmit_raw:
         code: [1028, -558, 604, -1193, 577, -559, 605, -876, 576, -1193, 579, -556, 607, -555, 604, -878, 579, -875, 577, -1193, 578, -1482, 577, -13982, 1028, -557, 605, -1194, 576, -558, 605, -875, 577, -1193, 579, -558, 605, -558, 604, -876, 577, -876, 577, -1193, 577, -1484, 578, -13981, 1027, -557, 605, -1194, 576, -560, 602, -877, 578, -1193, 577, -558, 604, -559, 605, -876, 576, -877, 577, -1193, 577, -1484, 578, -13981, 1026, -559, 604, -1194, 577, -558, 604, -877, 577, -1192, 577, -559, 604, -558, 604, -878, 576, -876, 577, -1193, 576, -1485, 556, -14003, 1027, -559, 603, -1194, 577, -559, 603, -877, 577, -1193, 576, -560, 604, -559, 601, -879, 578, -852, 601, -1192, 577, -1484, 555, -14005, 1025, -560, 603, -1194, 577, -559, 604, -853, 599, -1194, 576, -560, 604, -559, 603, -852, 601, -854, 600, -1193, 577, -1485, 576, -13982, 1026, -560, 602, -1195, 576, -559, 602, -857, 598, -1193, 577, -559, 582, -581, 582, -898, 554, -876, 577, -1217, 576, -1485, 554]
         carrier_frequency: 38kHz
  - platform: template
    id: mi_tv_dn
    name: "Mi TV Down"
    turn_on_action:
     - remote_transmitter.transmit_raw:
         code: [1026, -561, 580, -1213, 557, -582, 602, -876, 556, -1213, 558, -583, 579, -583, 580, -896, 557, -1214, 557, -1214, 556, -582, 581, -13104, 1027, -559, 583, -1213, 557, -580, 604, -875, 578, -1193, 580, -558, 606, -554, 608, -848, 605, -1189, 581, -1190, 583, -555, 606, -13079, 1033, -553, 608, -1189, 582, -553, 610, -870, 584, -1187, 582, -555, 608, -556, 607, -870, 582, -1188, 584, -1187, 584, -553, 610, -13076, 1034, -553, 633, -1162, 608, -529, 633, -846, 608, -1162, 585, -552, 636, -525, 612, -869, 609, -1138, 608, -1186, 608, -529, 610, -13075, 1058, -530, 633, -1162, 609, -528, 633, -822, 631, -1162, 609, -530, 632, -530, 632, -823, 631, -1138, 631, -1138, 634, -530, 632, -13054, 1055, -531, 632, -1138, 631, -532, 631, -821, 632, -1138, 631, -532, 631, -532, 630, -824, 630, -1140, 629, -1141, 628, -534, 630, -13058, 1026, -560, 603, -1166, 604, -559, 603, -851, 603, -1167, 602, -561, 602, -560, 601, -853, 601, -1169, 601, -1169, 601, -562, 600]
         carrier_frequency: 38kHz
  - platform: template
    id: mi_tv_lf
    name: "Mi TV Left"
    turn_on_action:
     - remote_transmitter.transmit_raw:
         code: [1004, -583, 579, -1215, 555, -584, 579, -899, 554, -1216, 555, -583, 579, -582, 581, -1215, 555, -1505, 556, -899, 555, -898, 554, -12524, 1005, -581, 583, -1214, 556, -582, 580, -898, 556, -1214, 556, -581, 581, -583, 580, -1215, 555, -1506, 556, -898, 555, -898, 556, -12522, 1006, -582, 581, -1214, 556, -582, 581, -897, 556, -1214, 579, -559, 604, -558, 582, -1213, 581, -1481, 578, -876, 555, -898, 556, -12523, 1029, -556, 608, -1189, 581, -557, 605, -850, 604, -1190, 581, -555, 607, -556, 606, -1190, 582, -1479, 583, -871, 582, -848, 606, -12496, 1031, -555, 611, -1186, 583, -554, 634, -819, 609, -1187, 583, -553, 610, -554, 633, -1162, 584, -1477, 610, -819, 609, -869, 584, -12495, 1057, -529, 636, -1160, 609, -528, 636, -844, 608, -1161, 609, -530, 633, -529, 633, -1161, 610, -1428, 633, -820, 632, -823, 630, -12472, 1057, -529, 632, -1139, 634, -529, 632, -821, 632, -1137, 633, -531, 631, -530, 632, -1139, 631, -1430, 631, -822, 631, -823, 630]
         carrier_frequency: 38kHz
  - platform: template
    id: mi_tv_rg
    name: "Mi TV Right"
    turn_on_action:
     - remote_transmitter.transmit_raw:
         code: [1029, -557, 606, -1192, 578, -558, 604, -876, 578, -1192, 578, -558, 604, -559, 604, -1483, 578, -558, 605, -558, 604, -1192, 579, -13082, 1028, -558, 604, -1193, 577, -559, 604, -852, 602, -1191, 579, -559, 604, -557, 605, -1483, 578, -557, 605, -558, 605, -1192, 579, -13084, 1026, -557, 606, -1192, 578, -558, 604, -851, 601, -1194, 578, -556, 606, -559, 603, -1484, 578, -557, 605, -559, 604, -1192, 577, -13085, 1025, -560, 602, -1195, 575, -561, 602, -854, 577, -1217, 576, -559, 602, -562, 602, -1485, 576, -561, 602, -559, 582, -1216, 577, -13084, 1026, -558, 606, -1192, 579, -558, 604, -851, 603, -1191, 579, -557, 606, -557, 607, -1481, 580, -556, 606, -556, 607, -1190, 581, -13079, 1032, -554, 632, -1141, 631, -529, 632, -823, 631, -1138, 633, -529, 634, -529, 632, -1429, 632, -531, 632, -531, 631, -1138, 632, -13055, 1054, -532, 630, -1140, 630, -533, 628, -825, 628, -1142, 604, -559, 604, -559, 627, -1434, 602, -561, 602, -560, 602, -1169, 601]
         carrier_frequency: 38kHz
  - platform: template
    id: mi_tv_ok
    name: "Mi TV OK"
    turn_on_action:
     - remote_transmitter.transmit_raw:
         code: [1052, -533, 629, -1168, 602, -534, 629, -824, 629, -1169, 596, -540, 628, -534, 629, -1458, 603, -824, 628, -535, 629, -1458, 602, -12478, 1051, -534, 629, -1168, 603, -533, 628, -826, 628, -1168, 602, -535, 607, -555, 612, -1476, 602, -824, 628, -535, 629, -1458, 602, -12478, 1050, -535, 605, -1193, 601, -534, 629, -824, 629, -1168, 602, -535, 628, -534, 629, -1458, 580, -847, 629, -534, 628, -1459, 604, -12477, 1051, -533, 628, -1169, 582, -554, 629, -824, 629, -1168, 602, -534, 628, -535, 629, -1457, 603, -826, 628, -534, 628, -1458, 594, -12488, 1045, -539, 607, -1189, 603, -534, 607, -847, 628, -1167, 591, -547, 628, -534, 604, -1481, 581, -850, 604, -557, 605, -1480, 581, -12502, 1026, -558, 605, -1190, 581, -557, 605, -850, 604, -1189, 581, -558, 628, -534, 605, -1480, 581, -850, 603, -558, 605, -1479, 582, -12501, 1028, -557, 605, -1189, 582, -557, 606, -848, 605, -1166, 603, -558, 605, -558, 605, -1457, 604, -850, 604, -558, 604, -1457, 605]
         carrier_frequency: 38kHz
  - platform: template
    id: mi_tv_back
    name: "Mi TV Back"
    turn_on_action:
     - remote_transmitter.transmit_raw:
         code: [1028, -557, 606, -1192, 578, -558, 604, -851, 602, -1193, 576, -559, 605, -558, 605, -851, 602, -1484, 577, -1193, 578, -850, 602, -12502, 1027, -558, 605, -1192, 578, -559, 603, -853, 601, -1193, 577, -558, 605, -558, 604, -851, 602, -1461, 601, -1193, 577, -850, 604, -12501, 1026, -559, 605, -1193, 577, -558, 605, -849, 604, -1170, 600, -558, 605, -558, 605, -850, 604, -1456, 605, -1166, 605, -848, 605, -12500, 1030, -556, 607, -1163, 630, -532, 632, -822, 631, -1139, 631, -532, 631, -532, 629, -824, 605, -1456, 605, -1166, 627, -826, 604, -12502, 1025, -560, 603, -1168, 602, -560, 602, -852, 600, -1170, 600, -562, 601, -562, 600, -854, 600, -1462, 598, -1192, 578, -876, 577, -12529, 999, -587, 576, -1194, 575, -588, 574, -879, 575, -1196, 574, -589, 573, -590, 572, -882, 570, -1491, 570, -1201, 568, -885, 566, -12565, 964, -598, 565, -1229, 541, -599, 564, -912, 541, -1229, 542, -621, 542, -621, 541, -912, 541, -1520, 541, -1230, 541, -912, 541]
         carrier_frequency: 38kHz
  - platform: template
    id: mi_tv_home
    name: "Mi TV Home"
    turn_on_action:
     - remote_transmitter.transmit_raw:
         code: [1051, -535, 628, -1167, 604, -534, 628, -824, 629, -1168, 603, -534, 628, -534, 629, -1168, 602, -535, 605, -847, 629, -1167, 603, -13059, 1051, -535, 629, -1167, 601, -536, 628, -825, 628, -1167, 604, -534, 628, -535, 608, -1187, 603, -534, 621, -831, 630, -1168, 593, -13068, 1051, -534, 628, -1168, 600, -537, 606, -849, 604, -1191, 580, -556, 606, -557, 610, -1186, 581, -557, 628, -825, 605, -1190, 580, -13082, 1028, -557, 606, -1190, 580, -557, 606, -849, 604, -1190, 580, -558, 605, -557, 605, -1190, 581, -557, 605, -850, 604, -1190, 580, -13083, 1028, -557, 604, -1190, 581, -558, 605, -849, 603, -1168, 604, -556, 605, -558, 605, -1166, 605, -558, 605, -850, 603, -1166, 605, -13082, 1053, -532, 617, -1153, 630, -533, 629, -825, 628, -1142, 604, -558, 605, -558, 603, -1167, 603, -560, 602, -851, 603, -1168, 602, -13085, 1024, -561, 601, -1171, 600, -561, 601, -854, 598, -1173, 597, -564, 599, -565, 597, -1193, 577, -566, 596, -878, 576, -1195, 575, -13112, 997, -589, 573, -1198, 572, -591, 571, -882, 571, -1200, 569, -594, 567, -596, 566, -1206, 564, -598, 565, -889, 564, -1229, 541, -13147, 964, -622, 540, -1230, 540, -623, 541, -912, 541, -1230, 541, -621, 541, -622, 541, -1230, 541, -622, 541, -912, 541, -1229, 541, -13173, 913, -671, 491, -1281, 489, -673, 490, -963, 490, -1281, 490, -673, 489, -673, 489, -1282, 489, -674, 489, -990, 463, -1307, 463, -13252, 833, -753, 409, -1387, 383, -805, 358, -1096, 357, -1440, 330, -860, 302, -1008, 103]
         carrier_frequency: 38kHz
  - platform: template
    id: mi_tv_volup
    name: "Mi TV Vol. Up"
    turn_on_action:
     - remote_transmitter.transmit_raw:
         code: [1054, -532, 631, -1139, 631, -532, 631, -823, 629, -1141, 629, -534, 629, -533, 630, -1431, 629, -1142, 628, -534, 605, -557, 629, -13059, 1028, -557, 604, -1166, 629, -534, 605, -848, 604, -1167, 604, -558, 604, -559, 603, -1458, 604, -1166, 604, -559, 603, -560, 603, -13084, 1026, -560, 602, -1167, 603, -560, 603, -851, 602, -1168, 602, -561, 601, -561, 601, -1461, 601, -1169, 600, -562, 601, -562, 600, -13088, 1022, -564, 598, -1173, 597, -585, 577, -876, 577, -1194, 576, -586, 577, -586, 576, -1486, 575, -1196, 574, -588, 574, -589, 574, -13113, 996, -590, 572, -1199, 570, -593, 570, -884, 568, -1203, 566, -597, 565, -598, 565, -1496, 565, -1206, 564, -599, 564, -599, 563, -13147, 964, -621, 541, -1230, 541, -621, 542, -911, 542, -1229, 541, -622, 541, -621, 541, -1521, 540, -1230, 542, -620, 542, -621, 541, -13147, 964, -622, 541, -1229, 541, -622, 541, -912, 541, -1230, 540, -622, 541, -622, 540, -1521, 525, -1246, 538, -625, 514, -672, 492]
         carrier_frequency: 38kHz
  - platform: template
    id: mi_tv_voldn
    name: "Mi TV Vol. Down"
    turn_on_action:
     - remote_transmitter.transmit_raw:
         code: [1039, -546, 628, -1169, 602, -533, 629, -824, 629, -1169, 602, -534, 628, -536, 627, -1460, 601, -1459, 602, -534, 629, -823, 630, -12478, 1027, -558, 629, -1167, 602, -534, 629, -825, 615, -1182, 602, -534, 629, -534, 628, -1459, 583, -1478, 602, -534, 629, -824, 629, -12479, 1050, -534, 628, -1169, 601, -535, 628, -824, 630, -1168, 602, -534, 628, -535, 604, -1482, 589, -1473, 602, -534, 628, -826, 628, -12477, 1051, -534, 629, -1168, 602, -534, 629, -825, 605, -1192, 602, -534, 618, -545, 628, -1459, 601, -1459, 580, -557, 605, -849, 604, -12502, 1027, -558, 605, -1192, 578, -558, 605, -849, 604, -1192, 579, -557, 605, -557, 606, -1483, 578, -1482, 579, -558, 605, -848, 605, -12500, 1028, -558, 605, -1192, 578, -557, 606, -849, 604, -1192, 578, -558, 605, -558, 605, -1482, 578, -1483, 579, -557, 604, -850, 604, -12502, 1027, -558, 605, -1166, 603, -558, 606, -850, 603, -1166, 604, -558, 605, -557, 606, -1456, 606, -1455, 606, -556, 607, -848, 629]
         carrier_frequency: 38kHz
```

**Logs (if applicable):**
<!--
Please copy the debug log here. If possible, also connect to the ESP over USB and copy those logs into the backticks.
-->
Home Assistant:
```
2021-06-17 15:14:28 ERROR (MainThread) [homeassistant] Error doing job: Task exception was never retrieved
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/components/esphome/__init__.py", line 214, in on_login
    entity_infos, services = await cli.list_entities_services()
  File "/usr/local/lib/python3.8/site-packages/aioesphomeapi/client.py", line 137, in list_entities_services
    entities.append(cls(**kwargs))
  File "<attrs generated init aioesphomeapi.model.ClimateInfo>", line 9, in __init__
    self.supported_modes = __attr_converter_supported_modes(supported_modes)
  File "/usr/local/lib/python3.8/site-packages/aioesphomeapi/model.py", line 245, in _convert_climate_modes
    return [ClimateMode(val) for val in value]
  File "/usr/local/lib/python3.8/site-packages/aioesphomeapi/model.py", line 245, in <listcomp>
    return [ClimateMode(val) for val in value]
  File "/usr/local/lib/python3.8/enum.py", line 339, in __call__
    return cls.__new__(cls, value)
  File "/usr/local/lib/python3.8/enum.py", line 662, in __new__
    raise ve_exc
ValueError: 6 is not a valid ClimateMode
```
NodeMCU Debug:
```
INFO Reading configuration /config/esphome/mcu_dnevna_soba_transciever.yaml...
INFO Generating C++ source...
INFO Compiling app...
INFO Running:  platformio run -d /config/esphome/mcu_dnevna_soba_transciever
Processing mcu_dnevna_soba_transciever (board: nodemcuv2; framework: arduino; platform: platformio/espressif8266@2.6.2)
--------------------------------------------------------------------------------
HARDWARE: ESP8266 80MHz, 80KB RAM, 4MB Flash
PACKAGES: 
 - framework-arduinoespressif8266 3.20704.0 (2.7.4) 
 - tool-esptool 1.413.0 (4.13) 
 - tool-esptoolpy 1.20800.0 (2.8.0) 
 - toolchain-xtensa 2.40802.200502 (4.8.2)
Library Manager: Installing Update
Library Manager: Already installed, built-in library
Dependency Graph
|-- <ESPAsyncTCP-esphome> 1.2.3
|   |-- <ESP8266WiFi> 1.0
|-- <IRremoteESP8266> 2.7.18
|-- <ESPAsyncWebServer-esphome> 1.2.7
|   |-- <ESPAsyncTCP-esphome> 1.2.3
|   |   |-- <ESP8266WiFi> 1.0
|   |-- <Hash> 1.0
|   |-- <ESP8266WiFi> 1.0
|   |-- <ArduinoJson-esphomelib> 5.13.3
|-- <ESP8266WiFi> 1.0
|-- <ESP8266mDNS> 1.2
|   |-- <ESP8266WiFi> 1.0
|-- <AsyncMqttClient-esphome> 0.8.4
|   |-- <ESPAsyncTCP-esphome> 1.2.3
|   |   |-- <ESP8266WiFi> 1.0
|-- <ArduinoJson-esphomelib> 5.13.3
|-- <DNSServer> 1.1.1
|   |-- <ESP8266WiFi> 1.0
Compiling /data/mcu_dnevna_soba_transciever/.pioenvs/mcu_dnevna_soba_transciever/src/main.cpp.o
In file included from src/main.cpp:69:0:
src/gree_ir.h: In member function 'virtual void GreeAC::control(const esphome::climate::ClimateCall&)':
src/gree_ir.h:67:14: warning: enumeration value 'CLIMATE_MODE_HEAT_COOL' not handled in switch [-Wswitch]
       switch (climateMode)
              ^
src/gree_ir.h:67:14: warning: enumeration value 'CLIMATE_MODE_FAN_ONLY' not handled in switch [-Wswitch]
src/gree_ir.h:105:14: warning: enumeration value 'CLIMATE_FAN_ON' not handled in switch [-Wswitch]
       switch (fanMode)
              ^
src/gree_ir.h:105:14: warning: enumeration value 'CLIMATE_FAN_OFF' not handled in switch [-Wswitch]
src/gree_ir.h:105:14: warning: enumeration value 'CLIMATE_FAN_MIDDLE' not handled in switch [-Wswitch]
src/gree_ir.h:105:14: warning: enumeration value 'CLIMATE_FAN_FOCUS' not handled in switch [-Wswitch]
src/gree_ir.h:105:14: warning: enumeration value 'CLIMATE_FAN_DIFFUSE' not handled in switch [-Wswitch]
src/gree_ir.h:128:14: warning: enumeration value 'CLIMATE_SWING_BOTH' not handled in switch [-Wswitch]
       switch (swingMode)
              ^
src/gree_ir.h:128:14: warning: enumeration value 'CLIMATE_SWING_HORIZONTAL' not handled in switch [-Wswitch]
Linking /data/mcu_dnevna_soba_transciever/.pioenvs/mcu_dnevna_soba_transciever/firmware.elf
Building /data/mcu_dnevna_soba_transciever/.pioenvs/mcu_dnevna_soba_transciever/firmware.bin
Retrieving maximum program size /data/mcu_dnevna_soba_transciever/.pioenvs/mcu_dnevna_soba_transciever/firmware.elf
Checking size /data/mcu_dnevna_soba_transciever/.pioenvs/mcu_dnevna_soba_transciever/firmware.elf
RAM:   [=====     ]  49.0% (used 40176 bytes from 81920 bytes)
Flash: [=====     ]  46.0% (used 480136 bytes from 1044464 bytes)
Creating BIN file "/data/mcu_dnevna_soba_transciever/.pioenvs/mcu_dnevna_soba_transciever/firmware.bin" using "/root/.platformio/packages/framework-arduinoespressif8266/bootloaders/eboot/eboot.elf" and "/data/mcu_dnevna_soba_transciever/.pioenvs/mcu_dnevna_soba_transciever/firmware.elf"
========================= [SUCCESS] Took 49.49 seconds =========================
INFO Successfully compiled program.
INFO Resolving IP address of mcu_dnevna_soba_transciever.local
INFO  -> 192.168.1.18
INFO Uploading /data/mcu_dnevna_soba_transciever/.pioenvs/mcu_dnevna_soba_transciever/firmware.bin (484288 bytes)
Uploading: [============================================================] 100% Done...


INFO Waiting for result...
INFO OTA successful
INFO Successfully uploaded program.
INFO Starting log output from mcu_dnevna_soba_transciever.local using esphome API
INFO Connecting to mcu_dnevna_soba_transciever.local:6053 (192.168.1.18)
INFO Successfully connected to mcu_dnevna_soba_transciever.local
[15:27:47][I][app:105]: ESPHome version 1.19.0 compiled on Jun 17 2021, 15:26:39
[15:27:47][C][wifi:484]: WiFi:
[15:27:47][C][wifi:344]:   SSID: 

[redacted]
[15:27:47][C][wifi:345]:   IP Address: 192.168.1.18
[15:27:47][C][wifi:347]:   BSSID: 

[redacted]
[15:27:47][C][wifi:348]:   Hostname: 'mcu_dnevna_soba_transciever'
[15:27:47][C][wifi:352]:   Signal strength: -39 dB ▂▄▆█
[15:27:47][C][wifi:356]:   Channel: 6
[15:27:47][C][wifi:357]:   Subnet: 255.255.254.0
[15:27:47][C][wifi:358]:   Gateway: 192.168.0.1
[15:27:47][C][wifi:359]:   DNS1: 192.168.0.1
[15:27:47][C][wifi:360]:   DNS2: (IP unset)
[15:27:47][C][template.switch:058]: Template Switch 'Mi TV'
[15:27:47][C][template.switch:059]:   Restore State: NO
[15:27:47][C][template.switch:060]:   Optimistic: NO
[15:27:47][C][template.switch:058]: Template Switch 'Mi TV Up'
[15:27:47][C][template.switch:059]:   Restore State: NO
[15:27:47][C][template.switch:060]:   Optimistic: NO
[15:27:47][C][template.switch:058]: Template Switch 'Mi TV Down'
[15:27:47][C][template.switch:059]:   Restore State: NO
[15:27:47][C][template.switch:060]:   Optimistic: NO
[15:27:47][C][template.switch:058]: Template Switch 'Mi TV Left'
[15:27:47][C][template.switch:059]:   Restore State: NO
[15:27:47][C][template.switch:060]:   Optimistic: NO
[15:27:47][C][template.switch:058]: Template Switch 'Mi TV Right'
[15:27:47][C][template.switch:059]:   Restore State: NO
[15:27:47][C][template.switch:060]:   Optimistic: NO
[15:27:47][C][template.switch:058]: Template Switch 'Mi TV OK'
[15:27:47][C][template.switch:059]:   Restore State: NO
[15:27:47][C][template.switch:060]:   Optimistic: NO
[15:27:47][C][template.switch:058]: Template Switch 'Mi TV Back'
[15:27:47][C][template.switch:059]:   Restore State: NO
[15:27:47][C][template.switch:060]:   Optimistic: NO
[15:27:47][C][template.switch:058]: Template Switch 'Mi TV Home'
[15:27:47][C][template.switch:059]:   Restore State: NO
[15:27:47][C][template.switch:060]:   Optimistic: NO
[15:27:47][C][template.switch:058]: Template Switch 'Mi TV Vol. Up'
[15:27:47][C][template.switch:059]:   Restore State: NO
[15:27:47][C][template.switch:060]:   Optimistic: NO
[15:27:47][C][template.switch:058]: Template Switch 'Mi TV Vol. Down'
[15:27:47][C][template.switch:059]:   Restore State: NO
[15:27:47][C][template.switch:060]:   Optimistic: NO
[15:27:47][C][gpio.binary_sensor:015]: GPIO Binary Sensor 'ir_dsoba_button'
[15:27:47][C][gpio.binary_sensor:016]:   Pin: GPIO13 (Mode: INPUT, INVERTED)
[15:27:47][C][logger:189]: Logger:
[15:27:47][C][logger:190]:   Level: DEBUG
[15:27:47][C][logger:191]:   Log Baud Rate: 115200
[15:27:47][C][logger:192]:   Hardware UART: UART0
[15:27:47][C][remote_transmitter:018]: Remote Transmitter...
[15:27:47][C][remote_transmitter:019]:   Carrier Duty: 50%
[15:27:47][C][remote_transmitter:020]:   Pin: GPIO16 (Mode: OUTPUT)
[15:27:47][C][captive_portal:148]: Captive Portal:
[15:27:47][C][ota:029]: Over-The-Air Updates:
[15:27:47][C][ota:030]:   Address: mcu_dnevna_soba_transciever.local:8266
[15:27:47][C][ota:032]:   Using Password.
[15:27:47][C][api:095]: API Server:
[15:27:47][C][api:096]:   Address: mcu_dnevna_soba_transciever.local:6053
[15:27:47][C][mqtt:059]: MQTT:
[15:27:47][C][mqtt:061]:   Server Address: 192.168.0.201:1883 (192.168.0.201)
[15:27:47][C][mqtt:062]:   Username: 

[redacted]
[15:27:47][C][mqtt:063]:   Client ID: 

[redacted]
[15:27:47][C][mqtt:068]:   Topic Prefix: 'mcu_dnevna_soba_transciever'
[15:27:47][C][mqtt:070]:   Log Topic: 'mcu_dnevna_soba_transciever/debug'
[15:27:47][C][mqtt:073]:   Availability: 'mcu_dnevna_soba_transciever/status'
[15:27:47][C][homeassistant.binary_sensor:037]: Homeassistant Binary Sensor 'Mi TV Status'
[15:27:47][C][homeassistant.binary_sensor:038]:   Entity ID: 'binary_sensor.mi_tv_status'
[15:27:47][C][mqtt.switch:038]: MQTT Switch 'Mi TV': 
[15:27:47][C][mqtt.switch:039]:   State Topic: 'mcu_dnevna_soba_transciever/switch/mi_tv/state'
[15:27:47][C][mqtt.switch:039]:   Command Topic: 'mcu_dnevna_soba_transciever/switch/mi_tv/command'
[15:27:47][C][mqtt.switch:038]: MQTT Switch 'Mi TV Up': 
[15:27:47][C][mqtt.switch:039]:   State Topic: 'mcu_dnevna_soba_transciever/switch/mi_tv_up/state'
[15:27:47][C][mqtt.switch:039]:   Command Topic: 'mcu_dnevna_soba_transciever/switch/mi_tv_up/command'
[15:27:47][C][mqtt.switch:038]: MQTT Switch 'Mi TV Down': 
[15:27:47][C][mqtt.switch:039]:   State Topic: 'mcu_dnevna_soba_transciever/switch/mi_tv_down/state'
[15:27:47][C][mqtt.switch:039]:   Command Topic: 'mcu_dnevna_soba_transciever/switch/mi_tv_down/command'
[15:27:47][C][mqtt.switch:038]: MQTT Switch 'Mi TV Left': 
[15:27:47][C][mqtt.switch:039]:   State Topic: 'mcu_dnevna_soba_transciever/switch/mi_tv_left/state'
[15:27:47][C][mqtt.switch:039]:   Command Topic: 'mcu_dnevna_soba_transciever/switch/mi_tv_left/command'
[15:27:47][C][mqtt.switch:038]: MQTT Switch 'Mi TV Right': 
[15:27:47][C][mqtt.switch:039]:   State Topic: 'mcu_dnevna_soba_transciever/switch/mi_tv_right/state'
[15:27:47][C][mqtt.switch:039]:   Command Topic: 'mcu_dnevna_soba_transciever/switch/mi_tv_right/command'
[15:27:47][C][mqtt.switch:038]: MQTT Switch 'Mi TV OK': 
[15:27:47][C][mqtt.switch:039]:   State Topic: 'mcu_dnevna_soba_transciever/switch/mi_tv_ok/state'
[15:27:47][C][mqtt.switch:039]:   Command Topic: 'mcu_dnevna_soba_transciever/switch/mi_tv_ok/command'
[15:27:47][C][mqtt.switch:038]: MQTT Switch 'Mi TV Back': 
[15:27:47][C][mqtt.switch:039]:   State Topic: 'mcu_dnevna_soba_transciever/switch/mi_tv_back/state'
[15:27:47][C][mqtt.switch:039]:   Command Topic: 'mcu_dnevna_soba_transciever/switch/mi_tv_back/command'
[15:27:47][C][mqtt.switch:038]: MQTT Switch 'Mi TV Home': 
[15:27:47][C][mqtt.switch:039]:   State Topic: 'mcu_dnevna_soba_transciever/switch/mi_tv_home/state'
[15:27:47][C][mqtt.switch:039]:   Command Topic: 'mcu_dnevna_soba_transciever/switch/mi_tv_home/command'
[15:27:47][C][mqtt.switch:038]: MQTT Switch 'Mi TV Vol. Up': 
[15:27:47][C][mqtt.switch:039]:   State Topic: 'mcu_dnevna_soba_transciever/switch/mi_tv_vol_up/state'
[15:27:47][C][mqtt.switch:039]:   Command Topic: 'mcu_dnevna_soba_transciever/switch/mi_tv_vol_up/command'
[15:27:47][C][mqtt.switch:038]: MQTT Switch 'Mi TV Vol. Down': 
[15:27:47][C][mqtt.switch:039]:   State Topic: 'mcu_dnevna_soba_transciever/switch/mi_tv_vol_down/state'
[15:27:47][C][mqtt.switch:039]:   Command Topic: 'mcu_dnevna_soba_transciever/switch/mi_tv_vol_down/command'
[15:27:47][C][mqtt.binary_sensor:018]: MQTT Binary Sensor 'ir_dsoba_button':
[15:27:47][C][mqtt.binary_sensor:019]:   State Topic: 'mcu_dnevna_soba_transciever/binary_sensor/ir_dsoba_button/state'
[15:27:47][C][mqtt.binary_sensor:018]: MQTT Binary Sensor 'Mi TV Status':
[15:27:47][C][mqtt.binary_sensor:019]:   State Topic: 'mcu_dnevna_soba_transciever/binary_sensor/mi_tv_status/state'
[15:27:48][C][mqtt.binary_sensor:018]: MQTT Binary Sensor 'Dnevna Soba Gree light':
[15:27:48][C][mqtt.binary_sensor:019]:   State Topic: 'mcu_dnevna_soba_transciever/binary_sensor/dnevna_soba_gree_light/state'
```

**Additional information and things you've tried:**
When i open climate entity from home assistant it's displayed as Unavailable. But when i click on "Operation" then select a mode, for example "Cool" or "Heat", AC will accept the command from ESP device and will respond, but home assistant entity is still displayed as unavailable.



<!-- LEAVE THIS LINE AS-IS AND DON'T DELETE IT, OTHERWISE THE ISSUE WILL BE CLOSED AUTOMATICALLY. -->

--- a/esphome/components/bang_bang/bang_bang_climate.cpp
+++ b/esphome/components/bang_bang/bang_bang_climate.cpp
@@ -21,7 +21,7 @@ void BangBangClimate::setup() {
     restore->to_call(this).perform();
   } else {
     // restore from defaults, change_away handles those for us
-    this->mode = climate::CLIMATE_MODE_AUTO;
+    this->mode = climate::CLIMATE_MODE_HEAT_COOL;
     this->change_away_(false);
   }
 }
@@ -50,7 +50,7 @@ climate::ClimateTraits BangBangClimate::traits() {
   return traits;
 }
 void BangBangClimate::compute_state_() {
-  if (this->mode != climate::CLIMATE_MODE_AUTO) {
+  if (this->mode != climate::CLIMATE_MODE_HEAT_COOL) {
     // in non-auto mode, switch directly to appropriate action
     //  - HEAT mode -> HEATING action
     //  - COOL mode -> COOLING action

--- a/esphome/components/bang_bang/bang_bang_climate.cpp
+++ b/esphome/components/bang_bang/bang_bang_climate.cpp
@@ -41,7 +41,7 @@ void BangBangClimate::control(const climate::ClimateCall &call) {
 climate::ClimateTraits BangBangClimate::traits() {
   auto traits = climate::ClimateTraits();
   traits.set_supports_current_temperature(true);
-  traits.set_supports_auto_mode(true);
+  traits.set_supports_heat_cool_mode(true);
   traits.set_supports_cool_mode(this->supports_cool_);
   traits.set_supports_heat_mode(this->supports_heat_);
   traits.set_supports_two_point_target_temperature(true);

--- a/esphome/components/climate/climate.h
+++ b/esphome/components/climate/climate.h
@@ -159,7 +159,7 @@ struct ClimateDeviceRestoreState {
  *
  * The entire state of the climate device is encoded in public properties of the base class (current_temperature,
  * mode etc). These are read-only for the user and rw for integrations. The reason these are public
- * is for simple access to them from lambdas `if (id(my_climate).mode == climate::CLIMATE_MODE_AUTO) ...`
+ * is for simple access to them from lambdas `if (id(my_climate).mode == climate::CLIMATE_MODE_HEAT_COOL) ...`
  */
 class Climate : public Nameable {
  public:

--- a/esphome/components/climate/climate_traits.cpp
+++ b/esphome/components/climate/climate_traits.cpp
@@ -8,6 +8,8 @@ bool ClimateTraits::supports_mode(ClimateMode mode) const {
   switch (mode) {
     case CLIMATE_MODE_OFF:
       return true;
+    case CLIMATE_MODE_HEAT_COOL:
+      return this->supports_heat_cool_mode_;
     case CLIMATE_MODE_AUTO:
       return this->supports_auto_mode_;
     case CLIMATE_MODE_COOL:
@@ -31,6 +33,9 @@ void ClimateTraits::set_supports_two_point_target_temperature(bool supports_two_
   supports_two_point_target_temperature_ = supports_two_point_target_temperature;
 }
 void ClimateTraits::set_supports_auto_mode(bool supports_auto_mode) { supports_auto_mode_ = supports_auto_mode; }
+void ClimateTraits::set_supports_heat_cool_mode(bool supports_heat_cool_mode) {
+  supports_heat_cool_mode_ = supports_heat_cool_mode;
+}
 void ClimateTraits::set_supports_cool_mode(bool supports_cool_mode) { supports_cool_mode_ = supports_cool_mode; }
 void ClimateTraits::set_supports_heat_mode(bool supports_heat_mode) { supports_heat_mode_ = supports_heat_mode; }
 void ClimateTraits::set_supports_fan_only_mode(bool supports_fan_only_mode) {

--- a/esphome/components/climate/climate_traits.h
+++ b/esphome/components/climate/climate_traits.h
@@ -46,6 +46,7 @@ class ClimateTraits {
   bool get_supports_two_point_target_temperature() const;
   void set_supports_two_point_target_temperature(bool supports_two_point_target_temperature);
   void set_supports_auto_mode(bool supports_auto_mode);
+  void set_supports_heat_cool_mode(bool supports_heat_cool_mode);
   void set_supports_cool_mode(bool supports_cool_mode);
   void set_supports_heat_mode(bool supports_heat_mode);
   void set_supports_fan_only_mode(bool supports_fan_only_mode);
@@ -100,6 +101,7 @@ class ClimateTraits {
   bool supports_current_temperature_{false};
   bool supports_two_point_target_temperature_{false};
   bool supports_auto_mode_{false};
+  bool supports_heat_cool_mode_{false};
   bool supports_cool_mode_{false};
   bool supports_heat_mode_{false};
   bool supports_fan_only_mode_{false};

--- a/esphome/components/climate_ir/climate_ir.cpp
+++ b/esphome/components/climate_ir/climate_ir.cpp
@@ -9,7 +9,7 @@ static const char *const TAG = "climate_ir";
 climate::ClimateTraits ClimateIR::traits() {
   auto traits = climate::ClimateTraits();
   traits.set_supports_current_temperature(this->sensor_ != nullptr);
-  traits.set_supports_auto_mode(true);
+  traits.set_supports_heat_cool_mode(true);
   traits.set_supports_cool_mode(this->supports_cool_);
   traits.set_supports_heat_mode(this->supports_heat_);
   traits.set_supports_dry_mode(this->supports_dry_);

--- a/esphome/components/climate_ir_lg/climate_ir_lg.cpp
+++ b/esphome/components/climate_ir_lg/climate_ir_lg.cpp
@@ -39,7 +39,7 @@ void LgIrClimate::transmit_state() {
     send_swing_cmd_ = false;
     remote_state |= COMMAND_SWING;
   } else {
-    if (mode_before_ == climate::CLIMATE_MODE_OFF && this->mode == climate::CLIMATE_MODE_AUTO) {
+    if (mode_before_ == climate::CLIMATE_MODE_OFF && this->mode == climate::CLIMATE_MODE_HEAT_COOL) {
       remote_state |= COMMAND_ON_AI;
     } else if (mode_before_ == climate::CLIMATE_MODE_OFF && this->mode != climate::CLIMATE_MODE_OFF) {
       remote_state |= COMMAND_ON;
@@ -52,7 +52,7 @@ void LgIrClimate::transmit_state() {
         case climate::CLIMATE_MODE_HEAT:
           remote_state |= COMMAND_HEAT;
           break;
-        case climate::CLIMATE_MODE_AUTO:
+        case climate::CLIMATE_MODE_HEAT_COOL:
           remote_state |= COMMAND_AUTO;
           break;
         case climate::CLIMATE_MODE_DRY:
@@ -89,7 +89,7 @@ void LgIrClimate::transmit_state() {
       }
     }
 
-    if (this->mode == climate::CLIMATE_MODE_AUTO) {
+    if (this->mode == climate::CLIMATE_MODE_HEAT_COOL) {
       this->fan_mode = climate::CLIMATE_FAN_AUTO;
       // remote_state |= FAN_MODE_AUTO_DRY;
     }
@@ -128,7 +128,7 @@ bool LgIrClimate::on_receive(remote_base::RemoteReceiveData data) {
   if ((remote_state & COMMAND_MASK) == COMMAND_ON) {
     this->mode = climate::CLIMATE_MODE_COOL;
   } else if ((remote_state & COMMAND_MASK) == COMMAND_ON_AI) {
-    this->mode = climate::CLIMATE_MODE_AUTO;
+    this->mode = climate::CLIMATE_MODE_HEAT_COOL;
   }
 
   if ((remote_state & COMMAND_MASK) == COMMAND_OFF) {
@@ -138,7 +138,7 @@ bool LgIrClimate::on_receive(remote_base::RemoteReceiveData data) {
         this->swing_mode == climate::CLIMATE_SWING_OFF ? climate::CLIMATE_SWING_VERTICAL : climate::CLIMATE_SWING_OFF;
   } else {
     if ((remote_state & COMMAND_MASK) == COMMAND_AUTO)
-      this->mode = climate::CLIMATE_MODE_AUTO;
+      this->mode = climate::CLIMATE_MODE_HEAT_COOL;
     else if ((remote_state & COMMAND_MASK) == COMMAND_DRY_FAN)
       this->mode = climate::CLIMATE_MODE_DRY;
     else if ((remote_state & COMMAND_MASK) == COMMAND_HEAT) {
@@ -152,7 +152,7 @@ bool LgIrClimate::on_receive(remote_base::RemoteReceiveData data) {
       this->target_temperature = ((remote_state & TEMP_MASK) >> TEMP_SHIFT) + 15;
 
     // Fan Speed
-    if (this->mode == climate::CLIMATE_MODE_AUTO) {
+    if (this->mode == climate::CLIMATE_MODE_HEAT_COOL) {
       this->fan_mode = climate::CLIMATE_FAN_AUTO;
     } else if (this->mode == climate::CLIMATE_MODE_COOL || this->mode == climate::CLIMATE_MODE_HEAT ||
                this->mode == climate::CLIMATE_MODE_DRY) {

--- a/esphome/components/coolix/coolix.cpp
+++ b/esphome/components/coolix/coolix.cpp
@@ -70,7 +70,7 @@ void CoolixClimate::transmit_state() {
       case climate::CLIMATE_MODE_HEAT:
         remote_state |= COOLIX_HEAT;
         break;
-      case climate::CLIMATE_MODE_AUTO:
+      case climate::CLIMATE_MODE_HEAT_COOL:
         remote_state |= COOLIX_AUTO;
         break;
       case climate::CLIMATE_MODE_FAN_ONLY:
@@ -89,7 +89,7 @@ void CoolixClimate::transmit_state() {
       } else {
         remote_state |= COOLIX_FAN_TEMP_CODE;
       }
-      if (this->mode == climate::CLIMATE_MODE_AUTO || this->mode == climate::CLIMATE_MODE_DRY) {
+      if (this->mode == climate::CLIMATE_MODE_HEAT_COOL || this->mode == climate::CLIMATE_MODE_DRY) {
         this->fan_mode = climate::CLIMATE_FAN_AUTO;
         remote_state |= COOLIX_FAN_MODE_AUTO_DRY;
       } else {
@@ -197,7 +197,7 @@ bool CoolixClimate::on_receive(remote_base::RemoteReceiveData data) {
     if ((remote_state & COOLIX_MODE_MASK) == COOLIX_HEAT)
       this->mode = climate::CLIMATE_MODE_HEAT;
     else if ((remote_state & COOLIX_MODE_MASK) == COOLIX_AUTO)
-      this->mode = climate::CLIMATE_MODE_AUTO;
+      this->mode = climate::CLIMATE_MODE_HEAT_COOL;
     else if ((remote_state & COOLIX_MODE_MASK) == COOLIX_DRY_FAN) {
       if ((remote_state & COOLIX_FAN_MASK) == COOLIX_FAN_MODE_AUTO_DRY)
         this->mode = climate::CLIMATE_MODE_DRY;
@@ -207,7 +207,7 @@ bool CoolixClimate::on_receive(remote_base::RemoteReceiveData data) {
       this->mode = climate::CLIMATE_MODE_COOL;
 
     // Fan Speed
-    if ((remote_state & COOLIX_FAN_AUTO) == COOLIX_FAN_AUTO || this->mode == climate::CLIMATE_MODE_AUTO ||
+    if ((remote_state & COOLIX_FAN_AUTO) == COOLIX_FAN_AUTO || this->mode == climate::CLIMATE_MODE_HEAT_COOL ||
         this->mode == climate::CLIMATE_MODE_DRY)
       this->fan_mode = climate::CLIMATE_FAN_AUTO;
     else if ((remote_state & COOLIX_FAN_MIN) == COOLIX_FAN_MIN)

--- a/esphome/components/daikin/daikin.cpp
+++ b/esphome/components/daikin/daikin.cpp
@@ -77,7 +77,7 @@ uint8_t DaikinClimate::operation_mode_() {
     case climate::CLIMATE_MODE_HEAT:
       operating_mode |= DAIKIN_MODE_HEAT;
       break;
-    case climate::CLIMATE_MODE_AUTO:
+    case climate::CLIMATE_MODE_HEAT_COOL:
       operating_mode |= DAIKIN_MODE_AUTO;
       break;
     case climate::CLIMATE_MODE_FAN_ONLY:
@@ -131,7 +131,7 @@ uint8_t DaikinClimate::temperature_() {
   switch (this->mode) {
     case climate::CLIMATE_MODE_FAN_ONLY:
       return 0x32;
-    case climate::CLIMATE_MODE_AUTO:
+    case climate::CLIMATE_MODE_HEAT_COOL:
     case climate::CLIMATE_MODE_DRY:
       return 0xc0;
     default:
@@ -160,7 +160,7 @@ bool DaikinClimate::parse_state_frame_(const uint8_t frame[]) {
         this->mode = climate::CLIMATE_MODE_HEAT;
         break;
       case DAIKIN_MODE_AUTO:
-        this->mode = climate::CLIMATE_MODE_AUTO;
+        this->mode = climate::CLIMATE_MODE_HEAT_COOL;
         break;
       case DAIKIN_MODE_FAN:
         this->mode = climate::CLIMATE_MODE_FAN_ONLY;

--- a/esphome/components/fujitsu_general/fujitsu_general.cpp
+++ b/esphome/components/fujitsu_general/fujitsu_general.cpp
@@ -133,7 +133,7 @@ void FujitsuGeneralClimate::transmit_state() {
     case climate::CLIMATE_MODE_FAN_ONLY:
       SET_NIBBLE(remote_state, FUJITSU_GENERAL_MODE_NIBBLE, FUJITSU_GENERAL_MODE_FAN);
       break;
-    case climate::CLIMATE_MODE_AUTO:
+    case climate::CLIMATE_MODE_HEAT_COOL:
     default:
       SET_NIBBLE(remote_state, FUJITSU_GENERAL_MODE_NIBBLE, FUJITSU_GENERAL_MODE_AUTO);
       break;
@@ -344,7 +344,7 @@ bool FujitsuGeneralClimate::on_receive(remote_base::RemoteReceiveData data) {
       case FUJITSU_GENERAL_MODE_AUTO:
       default:
         // TODO: CLIMATE_MODE_10C is missing from esphome
-        this->mode = climate::CLIMATE_MODE_AUTO;
+        this->mode = climate::CLIMATE_MODE_HEAT_COOL;
         break;
     }
 

--- a/esphome/components/hitachi_ac344/hitachi_ac344.cpp
+++ b/esphome/components/hitachi_ac344/hitachi_ac344.cpp
@@ -12,7 +12,7 @@ void set_bits(uint8_t *const dst, const uint8_t offset, const uint8_t nbits, con
   uint8_t mask = UINT8_MAX >> (8 - ((nbits > 8) ? 8 : nbits));
   // Calculate the mask & clear the space for the data.
   // Clear the destination bits.
-  *dst &= ~(uint8_t)(mask << offset);
+  *dst &= ~(uint8_t) (mask << offset);
   // Merge in the data.
   *dst |= ((data & mask) << offset);
 }
@@ -155,7 +155,7 @@ void HitachiClimate::transmit_state() {
     case climate::CLIMATE_MODE_HEAT:
       set_mode_(HITACHI_AC344_MODE_HEAT);
       break;
-    case climate::CLIMATE_MODE_AUTO:
+    case climate::CLIMATE_MODE_HEAT_COOL:
       set_mode_(HITACHI_AC344_MODE_AUTO);
       break;
     case climate::CLIMATE_MODE_FAN_ONLY:
@@ -251,7 +251,7 @@ bool HitachiClimate::parse_mode_(const uint8_t remote_state[]) {
         this->mode = climate::CLIMATE_MODE_HEAT;
         break;
       case HITACHI_AC344_MODE_AUTO:
-        this->mode = climate::CLIMATE_MODE_AUTO;
+        this->mode = climate::CLIMATE_MODE_HEAT_COOL;
         break;
       case HITACHI_AC344_MODE_FAN:
         this->mode = climate::CLIMATE_MODE_FAN_ONLY;

--- a/esphome/components/hitachi_ac344/hitachi_ac344.cpp
+++ b/esphome/components/hitachi_ac344/hitachi_ac344.cpp
@@ -12,7 +12,7 @@ void set_bits(uint8_t *const dst, const uint8_t offset, const uint8_t nbits, con
   uint8_t mask = UINT8_MAX >> (8 - ((nbits > 8) ? 8 : nbits));
   // Calculate the mask & clear the space for the data.
   // Clear the destination bits.
-  *dst &= ~(uint8_t) (mask << offset);
+  *dst &= ~(uint8_t)(mask << offset);
   // Merge in the data.
   *dst |= ((data & mask) << offset);
 }

--- a/esphome/components/midea_ac/midea_climate.cpp
+++ b/esphome/components/midea_ac/midea_climate.cpp
@@ -167,7 +167,7 @@ climate::ClimateTraits MideaAC::traits() {
   traits.set_visual_min_temperature(17);
   traits.set_visual_max_temperature(30);
   traits.set_visual_temperature_step(0.5);
-  traits.set_supports_auto_mode(true);
+  traits.set_supports_heat_cool_mode(true);
   traits.set_supports_cool_mode(true);
   traits.set_supports_dry_mode(true);
   traits.set_supports_heat_mode(true);

--- a/esphome/components/midea_ac/midea_frame.cpp
+++ b/esphome/components/midea_ac/midea_frame.cpp
@@ -44,7 +44,7 @@ climate::ClimateMode PropertiesFrame::get_mode() const {
     return climate::CLIMATE_MODE_OFF;
   switch (this->pbuf_[12] >> 5) {
     case MIDEA_MODE_AUTO:
-      return climate::CLIMATE_MODE_AUTO;
+      return climate::CLIMATE_MODE_HEAT_COOL;
     case MIDEA_MODE_COOL:
       return climate::CLIMATE_MODE_COOL;
     case MIDEA_MODE_DRY:
@@ -61,7 +61,7 @@ climate::ClimateMode PropertiesFrame::get_mode() const {
 void PropertiesFrame::set_mode(climate::ClimateMode mode) {
   uint8_t m;
   switch (mode) {
-    case climate::CLIMATE_MODE_AUTO:
+    case climate::CLIMATE_MODE_HEAT_COOL:
       m = MIDEA_MODE_AUTO;
       break;
     case climate::CLIMATE_MODE_COOL:

--- a/esphome/components/mitsubishi/mitsubishi.cpp
+++ b/esphome/components/mitsubishi/mitsubishi.cpp
@@ -33,7 +33,7 @@ void MitsubishiClimate::transmit_state() {
     case climate::CLIMATE_MODE_HEAT:
       remote_state[6] = MITSUBISHI_HEAT;
       break;
-    case climate::CLIMATE_MODE_AUTO:
+    case climate::CLIMATE_MODE_HEAT_COOL:
       remote_state[6] = MITSUBISHI_AUTO;
       break;
     case climate::CLIMATE_MODE_OFF:

--- a/esphome/components/pid/pid_climate.cpp
+++ b/esphome/components/pid/pid_climate.cpp
@@ -20,7 +20,7 @@ void PIDClimate::setup() {
     restore->to_call(this).perform();
   } else {
     // restore from defaults, change_away handles those for us
-    this->mode = climate::CLIMATE_MODE_AUTO;
+    this->mode = climate::CLIMATE_MODE_HEAT_COOL;
     this->target_temperature = this->default_target_temperature_;
   }
 }
@@ -31,7 +31,7 @@ void PIDClimate::control(const climate::ClimateCall &call) {
     this->target_temperature = *call.get_target_temperature();
 
   // If switching to non-auto mode, set output immediately
-  if (this->mode != climate::CLIMATE_MODE_AUTO)
+  if (this->mode != climate::CLIMATE_MODE_HEAT_COOL)
     this->handle_non_auto_mode_();
 
   this->publish_state();
@@ -121,14 +121,14 @@ void PIDClimate::update_pid_() {
         // keep autotuner instance so that subsequent dump_configs will print the long result message.
       } else {
         value = res.output;
-        if (mode != climate::CLIMATE_MODE_AUTO) {
+        if (mode != climate::CLIMATE_MODE_HEAT_COOL) {
           ESP_LOGW(TAG, "For PID autotuner you need to set AUTO (also called heat/cool) mode!");
         }
       }
     }
   }
 
-  if (this->mode != climate::CLIMATE_MODE_AUTO) {
+  if (this->mode != climate::CLIMATE_MODE_HEAT_COOL) {
     this->handle_non_auto_mode_();
   } else {
     this->write_output_(value);

--- a/esphome/components/pid/pid_climate.cpp
+++ b/esphome/components/pid/pid_climate.cpp
@@ -39,7 +39,7 @@ void PIDClimate::control(const climate::ClimateCall &call) {
 climate::ClimateTraits PIDClimate::traits() {
   auto traits = climate::ClimateTraits();
   traits.set_supports_current_temperature(true);
-  traits.set_supports_auto_mode(true);
+  traits.set_supports_heat_cool_mode(true);
   traits.set_supports_two_point_target_temperature(false);
   traits.set_supports_cool_mode(this->supports_cool_());
   traits.set_supports_heat_mode(this->supports_heat_());

--- a/esphome/components/tcl112/tcl112.cpp
+++ b/esphome/components/tcl112/tcl112.cpp
@@ -47,7 +47,7 @@ void Tcl112Climate::transmit_state() {
 
   // Set mode
   switch (this->mode) {
-    case climate::CLIMATE_MODE_AUTO:
+    case climate::CLIMATE_MODE_HEAT_COOL:
       remote_state[6] &= 0xF0;
       remote_state[6] |= TCL112_AUTO;
       break;
@@ -204,7 +204,7 @@ bool Tcl112Climate::on_receive(remote_base::RemoteReceiveData data) {
         this->mode = climate::CLIMATE_MODE_FAN_ONLY;
         break;
       case TCL112_AUTO:
-        this->mode = climate::CLIMATE_MODE_AUTO;
+        this->mode = climate::CLIMATE_MODE_HEAT_COOL;
         break;
     }
   }

--- a/esphome/components/thermostat/climate.py
+++ b/esphome/components/thermostat/climate.py
@@ -227,7 +227,7 @@ async def to_code(config):
     await cg.register_component(var, config)
     await climate.register_climate(var, config)
 
-    auto_mode_available = CONF_HEAT_ACTION in config and CONF_COOL_ACTION in config
+    heat_cool_mode_available = CONF_HEAT_ACTION in config and CONF_COOL_ACTION in config
     two_points_available = CONF_HEAT_ACTION in config and (
         CONF_COOL_ACTION in config or CONF_FAN_ONLY_ACTION in config
     )
@@ -258,10 +258,10 @@ async def to_code(config):
         var.get_idle_action_trigger(), [], config[CONF_IDLE_ACTION]
     )
 
-    if auto_mode_available is True:
-        cg.add(var.set_supports_auto(True))
+    if heat_cool_mode_available is True:
+        cg.add(var.set_supports_heat_cool(True))
     else:
-        cg.add(var.set_supports_auto(False))
+        cg.add(var.set_supports_heat_cool(False))
 
     if CONF_COOL_ACTION in config:
         await automation.build_automation(

--- a/esphome/components/thermostat/thermostat_climate.cpp
+++ b/esphome/components/thermostat/thermostat_climate.cpp
@@ -79,6 +79,7 @@ climate::ClimateTraits ThermostatClimate::traits() {
   auto traits = climate::ClimateTraits();
   traits.set_supports_current_temperature(true);
   traits.set_supports_auto_mode(this->supports_auto_);
+  traits.set_supports_heat_cool_mode(this->supports_heat_cool_);
   traits.set_supports_cool_mode(this->supports_cool_);
   traits.set_supports_dry_mode(this->supports_dry_);
   traits.set_supports_fan_only_mode(this->supports_fan_only_);
@@ -130,7 +131,6 @@ climate::ClimateAction ThermostatClimate::compute_action_() {
       case climate::CLIMATE_MODE_OFF:
         target_action = climate::CLIMATE_ACTION_OFF;
         break;
-      case climate::CLIMATE_MODE_AUTO:
       case climate::CLIMATE_MODE_HEAT_COOL:
       case climate::CLIMATE_MODE_COOL:
       case climate::CLIMATE_MODE_HEAT:
@@ -435,6 +435,9 @@ ThermostatClimate::ThermostatClimate()
       swing_mode_vertical_trigger_(new Trigger<>()) {}
 void ThermostatClimate::set_hysteresis(float hysteresis) { this->hysteresis_ = hysteresis; }
 void ThermostatClimate::set_sensor(sensor::Sensor *sensor) { this->sensor_ = sensor; }
+void ThermostatClimate::set_supports_heat_cool(bool supports_heat_cool) {
+  this->supports_heat_cool_ = supports_heat_cool;
+}
 void ThermostatClimate::set_supports_auto(bool supports_auto) { this->supports_auto_ = supports_auto; }
 void ThermostatClimate::set_supports_cool(bool supports_cool) { this->supports_cool_ = supports_cool; }
 void ThermostatClimate::set_supports_dry(bool supports_dry) { this->supports_dry_ = supports_dry; }
@@ -522,6 +525,7 @@ void ThermostatClimate::dump_config() {
   }
   ESP_LOGCONFIG(TAG, "  Hysteresis: %.1f°C", this->hysteresis_);
   ESP_LOGCONFIG(TAG, "  Supports AUTO: %s", YESNO(this->supports_auto_));
+  ESP_LOGCONFIG(TAG, "  Supports HEAT/COOL: %s", YESNO(this->supports_heat_cool_));
   ESP_LOGCONFIG(TAG, "  Supports COOL: %s", YESNO(this->supports_cool_));
   ESP_LOGCONFIG(TAG, "  Supports DRY: %s", YESNO(this->supports_dry_));
   ESP_LOGCONFIG(TAG, "  Supports FAN_ONLY: %s", YESNO(this->supports_fan_only_));

--- a/esphome/components/thermostat/thermostat_climate.cpp
+++ b/esphome/components/thermostat/thermostat_climate.cpp
@@ -21,7 +21,7 @@ void ThermostatClimate::setup() {
     restore->to_call(this).perform();
   } else {
     // restore from defaults, change_away handles temps for us
-    this->mode = climate::CLIMATE_MODE_AUTO;
+    this->mode = climate::CLIMATE_MODE_HEAT_COOL;
     this->change_away_(false);
   }
   // refresh the climate action based on the restored settings
@@ -131,6 +131,7 @@ climate::ClimateAction ThermostatClimate::compute_action_() {
         target_action = climate::CLIMATE_ACTION_OFF;
         break;
       case climate::CLIMATE_MODE_AUTO:
+      case climate::CLIMATE_MODE_HEAT_COOL:
       case climate::CLIMATE_MODE_COOL:
       case climate::CLIMATE_MODE_HEAT:
         if (this->supports_cool_) {
@@ -321,7 +322,7 @@ void ThermostatClimate::switch_to_mode_(climate::ClimateMode mode) {
     case climate::CLIMATE_MODE_OFF:
       trig = this->off_mode_trigger_;
       break;
-    case climate::CLIMATE_MODE_AUTO:
+    case climate::CLIMATE_MODE_HEAT_COOL:
       // trig = this->auto_mode_trigger_;
       break;
     case climate::CLIMATE_MODE_COOL:
@@ -339,7 +340,7 @@ void ThermostatClimate::switch_to_mode_(climate::ClimateMode mode) {
     default:
       // we cannot report an invalid mode back to HA (even if it asked for one)
       //  and must assume some valid value
-      mode = climate::CLIMATE_MODE_AUTO;
+      mode = climate::CLIMATE_MODE_HEAT_COOL;
       // trig = this->auto_mode_trigger_;
   }
   assert(trig != nullptr);

--- a/esphome/components/thermostat/thermostat_climate.h
+++ b/esphome/components/thermostat/thermostat_climate.h
@@ -29,6 +29,7 @@ class ThermostatClimate : public climate::Climate, public Component {
   void set_hysteresis(float hysteresis);
   void set_sensor(sensor::Sensor *sensor);
   void set_supports_auto(bool supports_auto);
+  void set_supports_heat_cool(bool supports_heat_cool);
   void set_supports_cool(bool supports_cool);
   void set_supports_dry(bool supports_dry);
   void set_supports_fan_only(bool supports_fan_only);
@@ -113,6 +114,7 @@ class ThermostatClimate : public climate::Climate, public Component {
   /// A false value for any given attribute means that the controller has no such action
   /// (for example a thermostat, where only heating and not-heating is possible).
   bool supports_auto_{false};
+  bool supports_heat_cool_{false};
   bool supports_cool_{false};
   bool supports_dry_{false};
   bool supports_fan_only_{false};

--- a/esphome/components/toshiba/toshiba.cpp
+++ b/esphome/components/toshiba/toshiba.cpp
@@ -80,7 +80,7 @@ void ToshibaClimate::transmit_state() {
       mode = TOSHIBA_MODE_COOL;
       break;
 
-    case climate::CLIMATE_MODE_AUTO:
+    case climate::CLIMATE_MODE_HEAT_COOL:
     default:
       mode = TOSHIBA_MODE_AUTO;
   }
@@ -190,7 +190,7 @@ bool ToshibaClimate::on_receive(remote_base::RemoteReceiveData data) {
     case TOSHIBA_MODE_AUTO:
     default:
       /* Note: Dry and Fan-only modes are reported as Auto, as they are not supported yet */
-      this->mode = climate::CLIMATE_MODE_AUTO;
+      this->mode = climate::CLIMATE_MODE_HEAT_COOL;
   }
 
   /* Get the target temperature */

--- a/esphome/components/tuya/climate/tuya_climate.cpp
+++ b/esphome/components/tuya/climate/tuya_climate.cpp
@@ -13,7 +13,7 @@ void TuyaClimate::setup() {
       this->mode = climate::CLIMATE_MODE_OFF;
       if (datapoint.value_bool) {
         if (this->supports_heat_ && this->supports_cool_) {
-          this->mode = climate::CLIMATE_MODE_AUTO;
+          this->mode = climate::CLIMATE_MODE_HEAT_COOL;
         } else if (this->supports_heat_) {
           this->mode = climate::CLIMATE_MODE_HEAT;
         } else if (this->supports_cool_) {

--- a/esphome/components/whirlpool/whirlpool.cpp
+++ b/esphome/components/whirlpool/whirlpool.cpp
@@ -48,7 +48,7 @@ void WhirlpoolClimate::transmit_state() {
     this->powered_on_assumed = powered_on;
   }
   switch (this->mode) {
-    case climate::CLIMATE_MODE_AUTO:
+    case climate::CLIMATE_MODE_HEAT_COOL:
       // set fan auto
       // set temp auto temp
       // set sleep false
@@ -78,7 +78,7 @@ void WhirlpoolClimate::transmit_state() {
 
   // Temperature
   auto temp = (uint8_t) roundf(clamp(this->target_temperature, this->temperature_min_(), this->temperature_max_()));
-  remote_state[3] |= (uint8_t)(temp - this->temperature_min_()) << 4;
+  remote_state[3] |= (uint8_t) (temp - this->temperature_min_()) << 4;
 
   // Fan speed
   switch (this->fan_mode.value()) {
@@ -239,7 +239,7 @@ bool WhirlpoolClimate::on_receive(remote_base::RemoteReceiveData data) {
         this->mode = climate::CLIMATE_MODE_FAN_ONLY;
         break;
       case WHIRLPOOL_AUTO:
-        this->mode = climate::CLIMATE_MODE_AUTO;
+        this->mode = climate::CLIMATE_MODE_HEAT_COOL;
         break;
     }
   }

--- a/esphome/components/whirlpool/whirlpool.cpp
+++ b/esphome/components/whirlpool/whirlpool.cpp
@@ -78,7 +78,7 @@ void WhirlpoolClimate::transmit_state() {
 
   // Temperature
   auto temp = (uint8_t) roundf(clamp(this->target_temperature, this->temperature_min_(), this->temperature_max_()));
-  remote_state[3] |= (uint8_t) (temp - this->temperature_min_()) << 4;
+  remote_state[3] |= (uint8_t)(temp - this->temperature_min_()) << 4;
 
   // Fan speed
   switch (this->fan_mode.value()) {

--- a/esphome/components/yashima/yashima.cpp
+++ b/esphome/components/yashima/yashima.cpp
@@ -82,7 +82,7 @@ const uint32_t YASHIMA_CARRIER_FREQUENCY = 38000;
 climate::ClimateTraits YashimaClimate::traits() {
   auto traits = climate::ClimateTraits();
   traits.set_supports_current_temperature(this->sensor_ != nullptr);
-  traits.set_supports_auto_mode(true);
+  traits.set_supports_heat_cool_mode(true);
   traits.set_supports_cool_mode(this->supports_cool_);
   traits.set_supports_heat_mode(this->supports_heat_);
   traits.set_supports_two_point_target_temperature(false);

--- a/esphome/components/yashima/yashima.cpp
+++ b/esphome/components/yashima/yashima.cpp
@@ -139,7 +139,7 @@ void YashimaClimate::transmit_state_() {
 
   // Set mode
   switch (this->mode) {
-    case climate::CLIMATE_MODE_AUTO:
+    case climate::CLIMATE_MODE_HEAT_COOL:
       remote_state[0] |= YASHIMA_MODE_AUTO_BYTE0;
       remote_state[5] |= YASHIMA_MODE_AUTO_BYTE5;
       break;

